x-pack/plugins/session_view/public/components/process_tree_alerts_filter/index.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useCallback, useMemo } from 'react';
+import {
+  EuiContextMenuItem,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiContextMenuPanel,
+  EuiPopover,
+  EuiText,
+  EuiHorizontalRule,
+} from '@elastic/eui';
+import { DEFAULT_ALERT_FILTER_VALUE } from '../../../common/constants';
+import {
+  ProcessEventAlertCategory,
+  DefaultAlertFilterType,
+  AlertTypeCount,
+} from '../../../common/types/process_tree';
+import { useStyles } from './styles';
+
+export interface ProcessTreeAlertsFilterDeps {
+  totalAlertsCount: number;
+  alertTypeCounts: AlertTypeCount[];
+  filteredAlertsCount: number;
+  onAlertEventCategorySelected: (value: ProcessEventAlertCategory | DefaultAlertFilterType) => void;
+}
+
+export const ProcessTreeAlertsFilter = ({
+  totalAlertsCount,
+  alertTypeCounts,
+  filteredAlertsCount,
+  onAlertEventCategorySelected,
+}: ProcessTreeAlertsFilterDeps) => {
+  const { filterStatus, popover } = useStyles();
+
+  const [selectedProcessEventAlertCategory, setSelectedProcessEventAlertCategory] = useState<
+    ProcessEventAlertCategory | DefaultAlertFilterType
+  >(DEFAULT_ALERT_FILTER_VALUE);
+
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  const onSelectedProcessEventAlertCategory = useCallback(
+    (event) => {
+      const [_, selectedAlertEvent] = event.target.textContent.split(' ');
+      setSelectedProcessEventAlertCategory(selectedAlertEvent);
+      onAlertEventCategorySelected(selectedAlertEvent);
+      closePopover();
+    },
+    [onAlertEventCategorySelected]
+  );
+
+  const alertCountsStatusText = useMemo(() => {
+    return totalAlertsCount === filteredAlertsCount
+      ? totalAlertsCount
+      : `${filteredAlertsCount} of ${totalAlertsCount}`;
+  }, [totalAlertsCount, filteredAlertsCount]);
+
+  const doesMultipleAlertTypesExist = useMemo(() => {
+    // Check if alerts consist of at least two alert event types
+    const multipleAlertTypeCount = alertTypeCounts.reduce((sumOfAlertTypes, { count }) => {
+      if (count > 0) {
+        return (sumOfAlertTypes += 1);
+      }
+      return 0;
+    }, 0);
+    return multipleAlertTypeCount > 1;
+  }, [alertTypeCounts]);
+
+  const alertEventCategoryFilterMenuButton = (
+    <EuiButtonEmpty
+      data-test-subj="sessionView:sessionViewAlertDetailsEmptyFilterButton"
+      size="s"
+      iconType="arrowDown"
+      iconSide="right"
+      onClick={onButtonClick}
+    >
+      <EuiText size="s">View: {selectedProcessEventAlertCategory} alerts </EuiText>
```

Comment:
We'll need to localize this text. Probably the label part of the alert category as well.

Reply:
What do you mean by `localize` the text?

Reply:
For language translation.

see: session_view/public/components/session_view/translations.ts for an example. Or anywhere using ```<FormattedMessage>``` component.